detox/src/android/core/NativeElement.js

```diff
@@ -42,12 +42,31 @@ class NativeElement {
     return await new ActionInteraction(this._invocationManager, this._matcher, action, traceDescription).execute();
   }
 
-  async longPress() {
-    const action = new actions.LongPressAction();
-    const traceDescription = actionDescription.longPress();
+  async longPress(arg1, arg2) {
+    let duration = null;
+    let point = null;
+
+    if (arg1 !== undefined && typeof arg1 === 'number' && arg2 === undefined) {
+      duration = arg1;
+    } else if (arg1 !== undefined && this._checkIfPointIsValid(arg1) && arg2 === undefined) {
+      point = arg1;
+    } else if (arg1 !== undefined && typeof arg1 === 'number' && this._checkIfPointIsValid(arg2)) {
+      duration = arg1;
+      point = arg2;
+    } else if (arg1 !== undefined || arg2 !== undefined) {
+      throw new Error('longPress accepts either a duration (number) or a point ({x: number, y: number}) as ' +
+          'its first argument, and optionally a point as its second argument (if the first argument is a duration).');
+    }
+
+    const action = new actions.LongPressAction(duration, point && point.x, point && point.y);
+    const traceDescription = actionDescription.longPress(duration, point);
     return await new ActionInteraction(this._invocationManager, this._matcher, action, traceDescription).execute();
   }
 
+  _checkIfPointIsValid(point) {
```

Comment:
It doesn't sit well with me that iOS and Android param arg validation logic is written twice.
See `src/utils/assertArgument.js` we use in iOS API. I think you can come up with a common logic for both platforms in this case.

Reply:
done